gantry/entrypoint.sh

```diff
@@ -46,10 +46,23 @@ echo "
 mkdir -p "${{ RUNTIME_DIR }}"
 cd "${{ RUNTIME_DIR }}"
 
-if [[ -n "$GITHUB_TOKEN" ]]; then
-    gh repo clone "$GITHUB_REPO" .
-else
-    git clone "https://github.com/$GITHUB_REPO" .
+# `git clone` occasionally fails on some cirrascale clusters due to "dusty cables" or something,
+# so we retry a few times automatically.
+attempts=1
+until [ "$attempts" -eq 5 ]
+do
+    if [[ -n "$GITHUB_TOKEN" ]]; then
+        gh repo clone "$GITHUB_REPO" . && break
+    else
+        git clone "https://github.com/$GITHUB_REPO" . && break
+    fi
+    attempts=$((attempts+1)) 
+    sleep 0.1
```

Comment:
```suggestion
    sleep 30
```

I'm not sure what a good default is, but 30s seems like nothing in comparison to a multi-hour job.

Reply:
30 seconds could have a big impact for distributed replica jobs that are waiting on each other. How about 10 seconds?

Reply:
Sure - 0.1s is what I imagined for testing the retry logic :-)